### ISSUE TITLE
Error: Cannot find file: 'Tone.js'

### ISSUE THREAD

**Issue by fredguest** [tambien/Piano#15]
Can anyone help me with the correct installation process for this lib? I have tone.js installed and working, but when I run `yarn install @tonejs/piano`, it gets added to the `package.json`, but it seems to be looking for things inside `node_modules` that are not there. 

This is the error message:

`Cannot find file: 'Tone.js' does not match the corresponding name on disk: './node_modules/Tone/build/tone'.`

This is my package.json:
```
{
  "name": "dotcom",
  "version": "0.9.0",
  "private": true,
  "dependencies": {
    "@tonejs/piano": "^0.1.0",
    "fileapi": "^2.0.25",
    "firebase": "^6.2.4",
    "fuse.js": "^3.4.5",
    "ismobilejs": "^0.5.2",
    "lodash": "^4.17.14",
    "prop-types": "^15.7.2",
    "qs": "^6.7.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-redux": "^7.1.0",
    "react-router-dom": "^5.0.1",
    "react-scripts": "3.0.1",
    "react-scroll-parallax": "^2.1.0",
    "react-waypoint": "^9.0.1",
    "redux": "^4.0.4",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0",
    "styled-components": "^4.3.2",
    "tone": "^13.8.25",
    "uuid": "^3.3.2",
    "validator": "^10.11.0",
    "xml-js": "^1.6.11"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "stage": "env-cmd .env.development react-scripts build"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ]
}
```

**Comment by tambien** [comment]
the way that it's currently setup is that the piano has a peer dependency of tone@>=14.1.0. that version of Tone is still under active development so not really production ready. but you can give it a shot. the modules that are required for the piano work, but the rest of the modules might not.

**Comment by fredguest** [comment]
oh cool thanks for the help! will give that a shot tonight and report back.

**Comment by fredguest** [comment]
no luck, when i `yarn add tone@14.1.0` i get the same error: `Cannot find file: 'Tone.js' does not match the corresponding name on disk: './node_modules/Tone/build/tone'.`

when i `yarn add tone@14.1.18` i get a new error but it still doesn't run:

```
./node_modules/@tonejs/piano/build/Piano.js
Cannot find file: 'index.js' does not match the corresponding name on disk: './node_modules/Tone/build/esm/tone'.
```

**Comment by tambien** [comment]
Not sure what the problem is, but one thing to look into would be why it's trying to look for `index.js` in that directory. 

The directory should be something like (depending on your js bunlder):
`./node_modules/tone/build/Tone.js` or `./node_modules/tone/build/esm/index.js` 

your error says:
`./node_modules/Tone/build/tone`

Are you trying to import tone using that path?

**Comment by fredguest** [comment]
So currently I'm using Tone 13.8.25, the PolySynth specifically, and it works beautifully, but I would love to have a sampled piano sound instead. If I remove all references to Tone from the code and `yarn remove tone` I am back to a bank slate, no Tone in the app at all. Then I take the follwoing steps: 1) run `yarn add tone@14.1.0`, 2) run `yarn add @tonejs/piano` (which gives me v0.1.1 in package.json), 3) in the file in which I'd like to instantiate the piano, I add the line `import Piano from '@tonejs/piano';`, and 4) run `yarn start` to boot the app, I get the above errors.

So I'm not importing Tone directly at all, I'm just trying to import the default export of @tonejs/piano as `Piano`. Do I need to import Tone directly to use the piano?

**Comment by tambien** [comment]
if i'm understanding correctly, you won't be able to have two versions of Tone installed in your app. So you can't for example use both 13.8.25 next to v14. Make sure that you're not importing Tone.js anywhere else, and it's only used in the piano. 

Also more specifically the peerDependency version is `14.1.18`. I didn't remember the exact patch version when i had mentioned it earlier, so make sure you're using `tone@^14.1.18`.

**Comment by fredguest** [comment]
Sorry i made the description more confusing than it needs be, I am not trying to use two different version of Tone at once, I am completely removing `v13.8.25` and then installing `v14.1.0`, so there is only one version of Tone in the app at all, which is `v14.1.0`.

When I try with `v14.1.18` instead of `v14.1.0`, I still get the above errors.

In the README, you have an example of using the Piano as follows:
```
const piano = new Piano({
  velocities: 5
})
```

but there is no example of where `Piano` in the above code is coming from. What is the correct way to import that `Piano` object in the example above?

**Comment by tambien** [comment]
i think you should be able to do `import {Piano} from @tonejs/piano` if you're using es6 or typescript

**Comment by fredguest** [comment]
no luck, with tone v14.1.18 installed, and @tonejs/piano v0.1.1, this line of code:
```
import { Piano } from '@tonejs/piano';
```
produces the following error:
```
./node_modules/@tonejs/piano/build/Piano.js
Cannot find file: 'index.js' does not match the corresponding name on disk: './node_modules/Tone/build/esm/tone'.
```

**Comment by fredguest** [comment]
I have as bog standard an ES6 environment as you will find these days, just create-react-app https://github.com/facebook/create-react-app with no custom configs at all.

**Comment by tomspaven** [comment]
Hi @fredguest @tambien 

Did either of you ever get to the bottom of this?  Seem to be getting the same issuer with a bog standard CRA SPA.

**Comment by fredguest** [comment]
@tomspaven nope. ended up finding another library https://github.com/danigb/soundfont-player

**Comment by danigb** [comment]
I think I've fixed this issue in #16
